### PR TITLE
Support for SGX HW mode and proxies in docker(-compose) and related network-mgmt scripts

### PR DESCRIPTION
**Pull request by g2flyer**
- HW mode is essentially enabled when SGX_MODE is set to HW.  Containers which are SGX-MODE sensitive (e.g., peer, boilerplate, cc-builder) will have an "-hw" extension to distinguish from their SIM-mode sibling
- proxy support is mostly documentation (in short, you need a new docker-compose version). That said, it will also work with the standard ubuntu docker-compose iff you start with make; The new one you mainly need if you start from a fresh repo without build and start right away with the scripts (e.g., `./demo/client/scripting/scenario-run --bootstrap ...`)
- besides above, there is also some docker file cleanup and some small extensions of the demo docu and a reference thereof in the top README

### CODE REVIEW

File: utils/docker/cc-builder/Dockerfile [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 FROM hyperledger/fabric-private-chaincode-dev:latest
 
+ARG SGX_MODE=SIM
```
Comment (bvavala) [inline utils/docker/cc-builder/Dockerfile]: do you need both of these?

Comment (g2flyer) [reply utils/docker/cc-builder/Dockerfile]: You mean ARG and ENV. Yes, they have different effects. I wanted it persistent in the container to have it consistent on how the container was built, for which you need the ENV. and the ARG is the only way you can pass something at build ...

File: demo/chaincode/fpc/Makefile [hunk 2]
```diff
@@ -28,9 +28,21 @@ build: $(BUILD_DIR)
...
+
+HW_EXTENSION=$(shell if [ "${SGX_MODE}" = "HW" ]; then echo "-hw"; fi)
+
+FPC_DOCKER_NAMESPACE := hyperledger/fabric-private-chaincode
+FPC_DOCKER_CC_BUILDER_NAME = $(FPC_DOCKER_NAMESPACE)-cc-builder$(HW_EXTENSION)
+
 docker-build: clean
-	$(DOCKER) image inspect hyperledger/fabric-private-chaincode-cc-builder > /dev/null 2>&1 || { cd $(TOP)/utils/docker && make cc-builder; }
-	$(DOCKER) run -u $$(id -u):$$(id -g) -v ${PWD}:/project/src/github.com/hyperledger-labs/fabric-private-chaincode/demo/chaincode/fpc -w /project/src/github.com/hyperledger-labs/fabric-private-chaincode/demo/chaincode/fpc hyperledger/fabric-private-chaincode-cc-builder sh -c 'make build'
+	$(DOCKER) image inspect hyperledger/fabric-private-chaincode-cc-builder > /dev/null 2>&1 \
```
[the remark this answers is not in the document]
Comment (g2flyer) [reply demo/chaincode/fpc/Makefile]: Good catch, (although it really should have been $(FPC_DOCKER_CC_BUILDER_NAME) for uniformity reason)

File: demo/README.md [hunk 3]
```diff
@@ -59,7 +59,9 @@ startFPCAuctionNetwork.sh [options]
    compliant auction chaincode($FPC_PATH/demo/chaincode/fpc), register auction users, and bring up
    both the fabric-gatway & frontend UI. If the fabric-gateway and frontend UI docker images have
    not previously been built it will build them, otherwise the script will reuse the images already
-   existing. The FPC chaincode will not be built unless specified by the flag --build-cc.
+   existing. The FPC chaincode will not be built unless specified by the flag --build-cc.  
+   By calling the script with both build options, you will be able to run the demo without having
```
[the remark this answers is not in the document]
Comment (g2flyer) [reply demo/README.md]: should be fixed now in fixup commit. also formated --help message to fit into standard 80 column terminal ..

**Comment by mbrandenburger** [comment]
This looks good to me! Thanks Michael!

**Pull request merged**